### PR TITLE
feat: Added Book, Movie and Rental Integration tests

### PR DESCRIPTION
**Pull request by DFraczek**
# Integration Tests for Book, Movie, and Rental

## Description

In this PR, integration tests for the Book, Movie, and Rental entities were added to ensure that the functionalities are working correctly. These tests verify that various operations like creating, reading, updating, and deleting entities are functioning as expected, as well as testing the interactions between different components.

### 1. **Book Integration Tests**
- **Tested Book Creation**: Added tests to verify that a new book can be created through the `BookController` API.
- **Tested Book Retrieval**: Added tests to verify that a book can be retrieved by its ID.
- **Tested Book Update**: Added tests to ensure that updating a book's data works as expected.
- **Tested Book Deletion**: Added tests to verify that a book can be deleted correctly.
- **Tested Book Search by Author/Genre/Duration**: Added tests to retrieve books by their author, genre, and duration parameters.

### 2. **Movie Integration Tests**
- **Tested Movie Creation**: Added tests for creating movies through the `MovieController` API.
- **Tested Movie Retrieval**: Added tests to retrieve a movie by its ID.
- **Tested Movie Update**: Added tests to check that movie details can be updated.
- **Tested Movie Deletion**: Included tests for deleting a movie and verifying it no longer exists.
- **Tested Movie Search by Director/Genre/File Format/Duration**: Added tests to retrieve movies based on the director, genre, file format, and duration range.

### 3. **Rental Integration Tests**
- **Tested Rental Creation**: Added tests to check that a rental transaction can be created successfully.
- **Tested Rental Retrieval**: Added tests to verify that rental records can be fetched by user ID and item ID.
- **Tested Rental Deletion**: Included tests to ensure that rentals can be deleted correctly and that deleted rentals no longer exist in the system.

## Improvements
- The tests ensure that the API endpoints are functioning correctly with various input scenarios, verifying both successful and unsuccessful outcomes.

## Testing
- MockMvc was used to simulate HTTP requests and validate the responses from the API endpoints.
- Various test cases were written to handle edge cases, such as invalid data, empty results, and updating non-existent records.
- The tests cover different HTTP methods including `POST`, `GET`, `PUT`, and `DELETE`.

## Conclusion
These integration tests help ensure that the Book, Movie, and Rental components work as expected

### CODE REVIEW

File: library/src/test/java/org/pollub/library/integration/BookControllerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private Book testBook;
+
+    @BeforeEach
+    void setUp() {
+        bookRepository.deleteAll();
+
+        testBook = createTestBook();
```
Comment (blackspyek) [inline library/src/test/java/org/pollub/library/integration/BookControllerIntegrationTest.java]: If not all tests has to use this, do not use this method here

**Review by blackspyek — CHANGES_REQUESTED** [review]
Only one to change!

**Review by blackspyek — APPROVED** [review]
LGTM
